lib/Basic/Unix/TaskQueue.inc

```diff
@@ -253,15 +255,17 @@ static bool readFromAPipe(int Pipe, std::string &Output) {
     }
 
     Output.append(outputBuffer, readBytes);
+    if (!untilEnd)
+      break;
```

Comment:
It looks like this now reads only 1k at a time normally. That seems a bit small according to my intuition. Rather than stop after one read, would it work better to stop whenever `readBytes != sizeof(outputBuffer)` (modulo EINTR)

Reply:
Wait--I had assumed the pipe was in non-blocking mode. Hmmm..... If blocking, you'd need to check that data are available before another read.

Reply:
Fair. I can extend the condition to `if (!(readBytes == sizeof(outputBuffer) || untilEnd))`, good enough?

Reply:
Well, the whole point here is to only read on fds that _are_ readable. Which, hm, yeah, is why I wrote it to be only one read per poll in the first place.

It's crude, but it works as-is and we're only digesting diagnostic streams here anyways, they're not exactly high volume.

Reply:
OK. We can tweak later.